### PR TITLE
Documentation: Fix typos and wording problems

### PR DESCRIPTION
**Pull request by RobertLarsen**

### CODE REVIEW

File: Documentation/teaching/lectures/debugging.rst [hunk 1]
```diff
@@ -612,14 +612,6 @@ KASan vs SLUB_DEBUG
      allocation/freeing of object. KASan catch bugs right before it will happen,
      so we always know exact place of first bad read/write.
 
-  * SLUB_DEBUG has lower overhead than KASan.
```
Comment (dbaluta) [inline Documentation/teaching/lectures/debugging.rst]: Not sure what exactly is this commit here, but it doesn't look good. Please always rebase your changes on top of latest master branch.

File: Documentation/teaching/labs/kernel_modules.rst [hunk 2]
```diff
@@ -974,7 +974,7 @@ copy the module to the VM, by running the following commands in
   $ make build
   $ make copy
```
Comment (dbaluta) [inline Documentation/teaching/labs/kernel_modules.rst:975]: Changes look good to me. One minor comment: Please always add a prefix to the commit message stating the kernel 'subsystem' touched by the change.

Also, please use a verb in the commit message. For this commit, we should have something like this:

```
Documentation: Fix a wording problem
```

File: Documentation/teaching/lectures/debugging.rst (hunk 1 above)
Comment (RobertLarsen) [reply Documentation/teaching/lectures/debugging.rst]: I'm not sure either. Master does not have that line either.
I'll undo this...somehow.

File: Documentation/teaching/labs/kernel_modules.rst (hunk 2 above)
Comment (RobertLarsen) [reply Documentation/teaching/labs/kernel_modules.rst:975]: Understood

**Comment by tavip** [comment]
Looks good to me as well. Since these are simple fixes, please squash all commits into one. Thanks!

File: Documentation/teaching/labs/device_drivers.rst [hunk 3]
```diff
@@ -479,7 +479,7 @@ Read and write
 ==============
 
 The read and write operations are reaching the device driver as a
-result of a userpsace program calling the read of write system calls:
+result of a userpsace program calling the read or write system calls:
```
Comment (razvand) [inline Documentation/teaching/labs/device_drivers.rst:482]: If you're at it, fix `userpsace` to `userspace` or `user space`.

File: Documentation/teaching/labs/introduction.rst [hunk 4]
```diff
@@ -413,7 +413,7 @@ to have a more complete understanding of a certain aspect.
 
 The main advantages of the Linux kernel are the access to sources and
 the open development system. Because of this, the Internet offers a
-larger number of documentation for kernel.
+larger number of documentation for the kernel.
```
Comment (razvand) [inline Documentation/teaching/labs/introduction.rst:416]: ... larger number of sources of information on the kernel.

File: Documentation/teaching/lectures/syscalls.rst [hunk 5]
```diff
@@ -280,7 +280,7 @@ read system call then it can corrupt kernel memory.
 
 
 Likewise, if a pointer passed by the application is invalid
-(e.g. unmaped, read-only for cases where its need to be written), it
+(e.g. unmapped, read-only for cases where its need to be written), it
```
Comment (razvand) [inline Documentation/teaching/lectures/syscalls.rst:283]: ... read-only for situations where it is used for writing) ...

**Comment by dbaluta** [comment]
@RobertLarsen thanks, looks good to me.

**Pull request merged**